lib/types/time-uuid.js

```diff
@@ -76,9 +83,59 @@ util.inherits(TimeUuid, Uuid);
  * If not provided, a random nodeId will be generated.
  * @param {String|Buffer} [clockId] A 2-length Buffer or string of 6 ascii characters representing the clock identifier.
  * If not provided a random clockId will be generated.
+ * @param {Function} [callback] An optional callback to be invoked with the error as first parameter and the created
+ * <code>TimeUuid</code> as second parameter. When a callback is provided, the random portions of the
+ * <code>TimeUuid</code> instance are created asynchronously.
+ * <p>
+ *   When nodeId and/or clockId portions are not provided, this method will generate them using
```

Comment:
We should also indicate this in the `TimeUuid` constructor since that can be used directly, and we could indicate to use `fromDate` or `now` if you desire for these values to be created asynchronously.

Reply:
Great catch! I'll add it to the constructor as well.